### PR TITLE
Document Runtime upgrade and rollback with Runtime Provisioner for devs

### PR DESCRIPTION
**Pull request by majakurcius**
**Description**

Changes proposed in this pull request:

Add developer documentation on: 
- the `upgradeRuntime` mutation performing Runtime upgrade by one release. 
- the `rollBackUpgradeOperation` mutation performing upgrade rollback.

**Related issue(s)**
#647 

**Related PR(s)**
#1153

### CODE REVIEW

File: docs/internal/runtime-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Runtime upgrade with the Runtime Provisioner
+
+## Upgrade Runtime
```
Comment (klaudiagrz) [inline docs/internal/runtime-upgrade.md:3]: Do we need this heading? I'd remove it and go on with the content.

File: docs/internal/runtime-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Runtime Provisioner allows to upgrade Kyma Runtime using the GraphQL API. To upgrade a Runtime to version 1.11.0 with the standard Runtime components use the following mutation:
```
Comment (klaudiagrz) [inline docs/internal/runtime-upgrade.md:5]:
```suggestion
Runtime Provisioner allows you to upgrade Kyma Runtime using the GraphQL API. To upgrade a Runtime to the 1.11.0 version with the standard Runtime components, use the following mutation:
```

File: docs/internal/runtime-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+}
+```
+
+The Kyma configuration passed to the upgrade mutation overrides the previous configuration. The components and parts of the configuration not specified in the new version are removed.
+
+Similarly to provisioning, the upgrade operation is asynchronous and returns the [Runtime Operation Status](../provisioner/08-03-runtime-operation-status.md).
+
+## Roll back the latest upgrade
+
+The Runtime Provisioner API allows to roll back the Kyma configuration to the one from before the latest upgrade. This mutation does not affect the cluster itself in any way, it only affects the database. In the case a cluster is rolled back manually, for example restored from a backup, rollback on the database must be performed to keep the data in sync. To roll back an upgrade, the upgrade must be the last operation performed by the Runtime Provisioner on the given Runtime. In other words, you cannot perform a rollback if other operations have been performed on the Runtime after the upgrade.
```
Comment (klaudiagrz) [inline docs/internal/runtime-upgrade.md:136]:
```suggestion
The Runtime Provisioner API allows you to roll back the Kyma configuration to the one from before the latest upgrade. This mutation does not affect the cluster itself in any way, it only affects the database. In the case a cluster is rolled back manually, for example restored from a backup, a rollback on the database must be performed to keep the data in sync. To roll back an upgrade, the upgrade must be the last operation performed by the Runtime Provisioner on the given Runtime. In other words, you cannot perform a rollback if other operations have been performed on the Runtime after the upgrade. 
```

File: docs/internal/runtime-upgrade.md (hunk 1 above)
Comment (majakurcius) [reply docs/internal/runtime-upgrade.md:3]: I think we do, cos there's another section on Rollback later.

File: docs/internal/runtime-upgrade.md (hunk 2 above)
Comment (majakurcius) [reply docs/internal/runtime-upgrade.md:5]: I'll add the comma, but I'll stick to the original version description as the one in the suggestion AFAIK is not really how it's written in the IT context.

**Pull request merged**